src/Algolia.Search/Utils/AlgoliaHelper.cs

```diff
@@ -101,6 +101,18 @@ internal static void EnsureObjectID<T>()
         /// <returns></returns>
         internal static string GetObjectID<T>(T data)
         {
+            Type itemType = typeof(T);
+
+            if (itemType == typeof(JObject))
+            {
+                JProperty objectID = JObject.FromObject(data).Property("objectID");
+
+                if (objectID != null && objectID.ToString() != "")
```

Comment:
cosmetic: you could have used String.IsNullOrEmpty(objectID.ToString()). Feel free to do it in a second PR.
reference: https://docs.microsoft.com/en-us/dotnet/api/system.string.isnullorempty?view=netcore-3.1